### ISSUE TITLE
Non-initialization actions cannot take input-file options

### ISSUE THREAD

**Issue by wthrowe** [sxs-collaboration/spectre#4210]
Initialization actions (those in `Phase::Initialization`) can take options from the input file by defining `initialization_tags` and `initialization_tags_to_keep`, but those aliases are not respected for actions outside that phase.  As far as I know, the only way to get an option is to define an initialization action that does nothing, but this has proven very bug-prone as it must be kept in sync in every executable.

**Comment by knelli2** [comment]
What is your goal here? Is it to have a simple tag and populate its value directly from options?

**Comment by wthrowe** [comment]
Yes.

**Comment by knelli2** [comment]
Is adding the tag to an existing initialization action's `initialization_tags` and `initialization_tags_to_keep` aliases in your component not an option?

**Comment by wthrowe** [comment]
That doesn't solve the problem with keeping the actions in sync.  (And probably actually makes it worse.)

**Comment by knelli2** [comment]
I guess I'm confused what you mean when you say "in-sync"? If an initialization action adds a tag (and keeps it), then all actions in subsequent phases have access to that tag. This is true for every exectuable that uses that initialization action without any extra work on the developers part.

Are you saying that that this shouldn't be the case and that non-initialization actions should be able to define tags that are created by options so that the tag is only included when the non-initialization action is included? Not the initialization action, which can be used in many other places and then we have a rogue/unused tag floating around because it was only needed for that one non-initialization action?

**Comment by wthrowe** [comment]
Yes.  And in particular this causes a silently ignored input file option.

**Comment by wthrowe** [comment]
Or, to clarify "Yes" to everything starting at "non-initialization actions should be able to define tags...".  The current behavior of initialization actions is fine.

**Comment by knelli2** [comment]
Hmm ok. How about this.

Non-initialization actions can (but don't have to) define an alias `simple_tags_created_from_options` where all tags in that list will just be created from options and don't need an initialization action. `Main` will find all these tags in these aliases from the non-initialization actions, add them to the collective `initialization_tags` and `initialization_tags_to_keep` for whatever component you are on, and then they will be created automatically.

An issue (or not depending on how you look at it) I see is that now you will have access to these tags in initialization actions without the tag being defined in the initialization action (now that we only have one Box vairant). So you can't really access this tag in the initialization action because sometimes it may not be there. Right now I don't see a way to protect against this.

**Comment by wthrowe** [comment]
Yes, that would solve it.  We should check if non-option simple and compute tags can be added too.

The initialization action access thing is already true of any tags requested by other initialization actions, or of global cache tags requested by non-initialization actions.

**Comment by knelli2** [comment]
So then the only time we would add simple tags to an initialization action is if we actually have to do some calculation in order to initialize the value of the non-option simple tag?

**Comment by wthrowe** [comment]
That could be true (although some widely used tags might just be added in one place), but it's actually quite common for the initialization actions to do non-default initialization.  Most grid quantities are allocated with the correct size, and all the time-related things are transformed from the input floating point values to slabs and slab fractions, for a few examples.

**Comment by knelli2** [comment]
Ahh ok makes sense. This all sounds good.

As I was thinking about this a bit more, it seems like it will only be useful if you actually have iterable actions in the `Phase::Evolve`. For the control system, interpolator, interpolation targets, observer/writer, memory monitor, etc.. (all the "idle" components which wait for simple actions to be called on them) we still have to use the initialization actions to initialize all tags. Not a reason to not make this change, but just something I realized.

**Comment by kidder** [comment]
based on discussions with @wthrowe in person today and @nilsdeppe previously:

It is on my to-do list to clean up some initialization things after going to a single data box in an distributed object:
- Most initialization actions will be replaced by mutators since all items are in the DataBox from the start, and mutators are easier to write/test than actions
- Will get rid of `initialization_items` and `initialization_items_to_keep` since there won't be (m)any initialization actions
- Split `simple_tags` defined by actions into `simple_tags_from_options` and `default_initialized_simple_tags`.  These along with `compute_tags` will be built from all the actions, components and metavariables as `simple_tags` and `compute_tags` are now.
- Add a RemoveItem action that resets the item to its default value and will trigger an ASSERT if accessed after it is removed. (This replaces the functionality of what `initialization_items_to_keep` currently is used for)

**Comment by nilsvu** [comment]
To answer your original question: if all you need is for an action to define an input-file option, then you can define `const_global_cache_tags` in the action. Of course these won't be stored in the DataBox as editable simple tags, but in the const global cache. Therefore, it's an easy way to define input-file options for constant parameters and such. From your follow-up comments I understand you want an editable simple tag that's default-initialized from options, but your original question didn't say that so I'm leaving this answer in case someone else finds this issue.
